polar-core/src/vm.rs

```diff
@@ -1721,7 +1722,7 @@ impl PolarVirtualMachine {
                     instance_id: right_instance,
                     ..
                 }),
-            ) if left_instance != right_instance => {
+            ) => {
```

Comment:
highlighting this change for visibility

Reply:
Oh thanks, I meant to ping @plotnick on this one. The reason you wouldn't want to outright skip the unify if the instance IDs are equals are in the case where the two classes are not self-equals ,e.g. `nan != nan`.

Reply:
maybe a counterpoint (and definitely splitting hairs): isn't this more of an "is this the same thing" check instead of an "are these two things equal" check?

Reply:
Hmm, that's true. And would be more consistent with our rationale of letting `==` be overloadable, but not `=`.